WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanListItemState.kt

```diff
@@ -0,0 +1,42 @@
+package org.wordpress.android.ui.jetpack.scan
+
+import androidx.annotation.DrawableRes
+import org.wordpress.android.R
+import org.wordpress.android.ui.jetpack.scan.ScanListItemState.ViewType.SCAN_STATE
+import org.wordpress.android.ui.utils.UiString
+import org.wordpress.android.ui.utils.UiString.UiStringRes
+
+sealed class ScanListItemState(val type: ViewType) {
+    open fun longId(): Long = hashCode().toLong()
+
+    // TODO: ashiagr fine-tune states, dynamic texts, add button states
+    sealed class ScanState : ScanListItemState(SCAN_STATE) {
+        @DrawableRes open val scanIcon: Int = R.drawable.ic_scan_scanning
+        open val scanTitle: UiString = UiStringRes(R.string.scan_scanning_title)
+        open val scanDescription: UiString = UiStringRes(R.string.scan_scanning_description)
```

Comment:
Question (❓ ): I am not sure if those parent fields will be used somehow in the future, if not, what if this is changed to being `abstract` instead?
```
        abstract val scanIcon: Int
        abstract val scanTitle: UiString
        abstract val scanDescription: UiString
```

Reply:
In a576c37